<launch>
    <arg name="soma_config"/>
    <arg name="time_increment" default="1"/>
    <arg name="time_window" default="10"/>
    <arg name="periodic_cycle" default="1440"/>
    <arg name="consent_topic" default="/skeleton_data/consent_ret"/>
    <arg name="trajectory_topic" default="/people_trajectory/trajectories/complete"/>
    <arg name="consent_rate" default="1.5"/>

    <arg name="machine" default="localhost" />
    <arg name="user" default="" />

    <machine name="$(arg machine)" address="$(arg machine)" env-loader="$(optenv ROS_ENV_LOADER )" user="$(arg user)" default="true"/>

    <node pkg="activity_exploration" type="counting_manager.py" name="people_count_manager" output="screen" respawn="true">
        <param name="soma_config" value="$(arg soma_config)" type="string"/>
        <param name="time_increment" value="$(arg time_increment)" type="int"/>
        <param name="time_window" value="$(arg time_window)" type="int"/>
        <param name="periodic_cycle" value="$(arg periodic_cycle)" type="int"/>
        <param name="consent_topic" value="$(arg consent_topic)" type="string"/>
        <param name="trajectory_topic" value="$(arg trajectory_topic)" type="string"/>
        <param name="consent_rate" value="$(arg consent_rate)" type="double"/>
    </node>
    <node pkg="activity_exploration" type="observe.py" name="region_observation" respawn="true">
        <param name="soma_config" value="$(arg soma_config)" type="string"/>
        <param name="time_increment" value="$(arg time_increment)" type="int"/>
    </node>
</launch>
